ftp.go

```diff
@@ -337,7 +338,7 @@ func (c *ServerConn) NameList(path string) (entries []string, err error) {
 		return
 	}
 
-	r := &Response{conn, c}
+	r := &Response{conn, c, false}
```

Comment:
You could consider using keyed fields instead:

```Go
r := &Response{conn: conn, c: c}
```

`connClosed` will have the zero bool value, which is `false`, as desired.